src/types/ScreenInfoUiaProviderBase.cpp

```diff
@@ -272,13 +276,17 @@ IFACEMETHODIMP ScreenInfoUiaProviderBase::GetSelection(_Outptr_result_maybenull_
 
 IFACEMETHODIMP ScreenInfoUiaProviderBase::GetVisibleRanges(_Outptr_result_maybenull_ SAFEARRAY** ppRetVal)
 {
+    RETURN_HR_IF_NULL(E_INVALIDARG, ppRetVal);
+    if (!_pData->IsUiaDataInitialized())
+    {
+        return E_FAIL;
+    }
+

```

Comment:
Isn't this on the wrong side of the lock? Or alternatively, shouldn't the value be atomic or something?

Reply:
Eh I guess if it comes back as a dirty "false" it only goes to "true" once and a missed early call is OK. So maybe nevermind.

Reply:
@lhecker can you comment on this before we merge? Do you think this is OK as is?

Reply:
Theoretically it's not okay. Practically it'll work on any modern CPU.
The reason it's not save theoretically is because the standard says:
> If a data race occurs, the behavior of the program is undefined.

So theoretically speaking, the moment you have a race condition your computer could turn into a mahou shoujo.

Reply:
So what's better to you? Moving it all past the existing call to the `til::ticket_lock` or using an `std::atomic<bool>` on this flag?